contents/staticAnalysis/FindBugs.md

```diff
@@ -1,16 +1,83 @@
 # FindBugs
 
-Author: [Xiao Pu](https://nus-oss.github.io/cs3281-website/students/AY1617S2/xiaoPu/xiaoPu-Resume.html)
+Author: [Xiao Pu](https://nus-oss.github.io/cs3281-website/students/AY1617S2/xiaoPu/xiaoPu-Resume.html), [Shradheya Thakre](https://github.com/tshradheya)
 
 ## Overview
 
-FindBugs is a static analysis tool to find "bugs" in **Java** programme. It looks for "bug patterns" in the code and signals possible violations. For example, the bug "null pointer dereference" has the pattern — A programme declares a non-nullable variable but assigns `null` to the variable somewhere and uses it later.
+FindBugs is a static analysis tool to find "bugs" in **Java** programme. It looks for "bug patterns" in the code and signals possible violations. Potential errors are classified in four ranks:
+1. scariest
+1. scary
+1. troubling
+1. of concern
+
+This is a hint to the developer about their possible impact or severity. For example, the bug "null pointer dereference" has the pattern — A programme declares a non-nullable variable but assigns `null` to the variable somewhere and uses it later.
 
 ## Features
 
-The "bug patterns" can be divided into nine groups: Bad practice, Correctness, Experimental, Internationalization, Malicious code vulnerability, Multithreaded correctness, Performance, Security and Dodgy code. [A comprehensive list](http://findbugs.sourceforge.net/bugDescriptions.html) of bugs is provided to explain the meaning of each bug.
+The "bug patterns" can be divided into nine groups:
+1. Bad practice
+1. Correctness
+1. Experimental
+1. Internationalization
+1. Malicious code vulnerability
+1. Multithreaded correctness
+1. Performance
+1. Security
+1. Dodgy code
+
+[A comprehensive list](http://findbugs.sourceforge.net/bugDescriptions.html) of bugs is provided to explain the meaning of each bug.
+
+FindBugs analyses bytecode in compiled Java `.class` file and checks multiple files at the same time. This is unlike [CheckStyle](checkStyle.md) or [PMD](PMD.md) which can only check files one by one and analyse Java source code, allowing FindBugs to spot errors that would have been missed by CheckStyle and PMD. For example, one of the bug patterns in FindBugs is `RCN: Redundant nullcheck of value known to be non-null`. FindBugs will analyse all the assignments to a particular variable in the code base and then check whether the `nullcheck` for the variable is redundant or not.
+
+## Examples of Bugs that can be found using FindBugs
+
+### Find hash-equals mismatch
```

Comment:
Find equals/hash-equals mismatch?

Your first point mainly talk about overriding equals method in a wrong way. This is not relevant to hash right? Therefore, I suggest either u extract it out and form a new section or change the title of the current section.